### ISSUE TITLE
scan tar file inside target

### ISSUE THREAD

**Issue by deitch** [anchore/syft#1379]
**What would you like to be added**:

Reading tar or tgz files (as well as some other kinds of compression) is fairly straightforward. If I have a tar file inside my target (e.g. an OCI Image that includes a tarball, or a directory with a tarball in it), the contents of the tar are ignored.

**Why is this needed**:

Software distributions, especially OCI images, often contain bundled software in tar format inside it. We should be able to capture that as well.

**Additional context**:

To recreate this:

1. Create a simple project, e.g. a golang project
2. Compile the project, save the output to a directory, e.g. `/tmp/pkg`; or do an apk install or anything else recognized by syft
3. Run `syft packages dir:/tmp/pkg` - it finds the binary and various dependencies
4. tar up the directory: `tar -C /tmp/pkg -cvf /tmp/pkg.tar .`
5. Place the tar file in another directory, e.g. `mkdir /tmp/other && mv pkg.tar /tmp/other/pkg.tar`
6. Run `syft packages dir:/tmp/other` - it finds zero dependencies

**Comment by spiffcs** [comment]
This is a great point @deitch - We're only doing this functionality specifically for jars within jars under the java cataloger.

We'll throw this directly into the backlog to pick up. 

The first case to inspect is the performance surrounding if we did this for all `.tgz` or other archives, but I agree that this should not be functionality reserved for just the Java ecosystem.

**Comment by deitch** [comment]
> We'll throw this directly into the backlog to pick up.

Thanks. I have not looked deeply into syft source, so I do not know if saying, "let me know if I can help" is dangerous, but I will ask anyways.

> The first case to inspect is the performance surrounding if we did this for all .tgz or other archives, but I agree that this should not be functionality reserved for just the Java ecosystem

At the very least, let it be an option?

**Comment by deitch** [comment]
Huh, I just realized that syft does not read a tar of a directory, e.g. `syft path/to/tarfile.tar` unless that is a container image archive. I will open a new issue for it.

**Comment by deitch** [comment]
Never mind, it does; it is a different issue. I will open an issue.

**Comment by jedevc** [comment]
I have a similar use case as @deitch: we bundle up buildkit binary releases into tarballs - these are all go binaries, that when scanned by themselves, syft detects the packaging information. But, when they're placed into a tar somewhere in the filesystem, they aren't picked up.

We're able to work around by taking a scan result before we do the tar bundling, but it would be wonderful if the default experience had someway of picking up the results after :tada:

**Comment by kzantow** [comment]
Yes, today the only archives that are expanded are Java archives `jar`, `war`, `ear`, etc.. Performance is a possible consideration, but assuming that doesn't prove too slow, I think we definitely would like to support extracting archives recursively. We've added this to the backlog 👍

**Comment by wagoodman** [comment]
This seems related/overlapping with https://github.com/anchore/syft/issues/246

**Comment by mithunms333** [comment]
requesting to kindly share if this is planned in any upcoming enhancement or bug-fix cycle? I am not from the tech stack of syft, and hence unable to fix it myself.
Since a high count of application images contain tars which get extracted during container startup/run time, cataloguing the contents of tar or .tgz files is very important. Else this forms a gap in SBOM contents. To begin with, the developer might choose to set a nested max depth of 2 or 3, and later see if necessary to go further deep.
